subsys/logging/log_output_syst.c

```diff
@@ -0,0 +1,221 @@
+	u32_t severity = level_to_syst_severity(log_msg_level_get(msg));
+
+	for (int i = 0; i < nargs; i++) {
+		args[i] = log_msg_arg_get(msg, i);
+	}
+
+	switch (log_msg_nargs_get(msg)) {
+	case 0:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str);
+		break;
+	case 1:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0]);
+		break;
+	case 2:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1]);
+		break;
+	case 3:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2]);
+		break;
+	case 4:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2], args[3]);
+		break;
+	case 5:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2], args[3], args[4]);
+		break;
+	case 6:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2], args[3], args[4], args[5]);
+		break;
+	case 7:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2], args[3], args[4], args[5],
+				args[6]);
+		break;
+	case 8:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2], args[3], args[4], args[5],
+				args[6], args[7]);
+		break;
+	case 9:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2], args[3], args[4], args[5],
+				args[6], args[7], args[8]);
+		break;
+	case 10:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2], args[3], args[4], args[5],
+				args[6], args[7], args[8], args[9]);
+		break;
+	case 11:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2], args[3], args[4], args[5],
+				args[6], args[7], args[8], args[9], args[10]);
+		break;
+	case 12:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2], args[3], args[4], args[5],
+				args[6], args[7], args[8], args[9], args[10],
+				args[11]);
+		break;
+	case 13:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2], args[3], args[4], args[5],
+				args[6], args[7], args[8], args[9], args[10],
+				args[11], args[12]);
+		break;
+	case 14:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2], args[3], args[4], args[5],
+				args[6], args[7], args[8], args[9], args[10],
+				args[11], args[12], args[13]);
+		break;
+	case 15:
+		MIPI_SYST_PRINTF(&log_syst_handle, severity, str, args[0],
+				args[1], args[2], args[3], args[4], args[5],
+				args[6], args[7], args[8], args[9], args[10],
+				args[11], args[12], args[13], args[14]);
+		break;
+	default:
+		/* Unsupported number of arguments. */
+		__ASSERT_NO_MSG(true);
+		break;
+	}
+}
+
+static void raw_string_print(struct log_msg *msg,
+			const struct log_output *log_output)
+{
+	char buf[CONFIG_LOG_STRDUP_MAX_STRING + 1];
+	size_t length = CONFIG_LOG_STRDUP_MAX_STRING;
+	u32_t severity = level_to_syst_severity(log_msg_level_get(msg));
```

Comment:
raw string has custom `LOG_LEVEL_INTERNAL_RAW_STRING` (5) here it will be converted to debug level. Not sure if that is an expected result. Raw string is used when printk goes through logger. Alternatively severity can be fixed here to 0.

Reply:
@nordic-krch yes, we use raw string if printk goes through logger, and also I have it tested in sample code like below.
```

        /* raw string */
        printk("hello sys-t on board %s\n", CONFIG_BOARD);
```
And checked code again, it seems `LOG_LEVEL_INTERNAL_RAW_STRING` already equals 0 as below marco defined. If anything wrong, please correct me. Thanks
`#define LOG_LEVEL_INTERNAL_RAW_STRING LOG_LEVEL_NONE`

Reply:
my bad, somehow i thought it had different number.